src/main/java/hudson/plugins/active_directory/AbstractActiveDirectoryAuthenticationProvider.java

```diff
@@ -45,13 +53,48 @@ protected AbstractActiveDirectoryAuthenticationProvider() {
      */
     protected abstract UserDetails retrieveUser(String username, UsernamePasswordAuthenticationToken authentication) throws AuthenticationException;
 
+    protected abstract GroupDetails retrieveGroup(String groupname);
+
     /**
      * Returns true if we can retrieve user just from the name without supplying any credential.
      */
     protected abstract boolean canRetrieveUserByName();
 
+    public GroupDetails loadGroupByGroupname(String groupname) {
+      if(
+            this.realm!=null && 
+            this.realm.getSsoEnabled()!=null && 
+            this.realm.getSsoEnabled().getCheckMode()!=null && 
+            this.realm.getSsoEnabled().getCheckMode().equalsIgnoreCase("no_check")==true && 
+            this.realm.getSsoEnabled().getDefaultGroup()!=null  && 
+            this.realm.getSsoEnabled().getDefaultGroup().equals(groupname)==true)    
```

Comment:
a matter of preference maybe, but when conditions get complex, I generally extract it to a small dedicated boolean method. Something like isBlahMode() or hasSomethingSpecificBluhEnabled().